### ISSUE TITLE
~Circuit negation of global constraint not working correctly

### ISSUE THREAD

**Issue by Wout4** [CPMpy/cpmpy#214]
```python3
iv = intvar(0,8, shape=3)
Model([~Circuit(iv), iv==[1,2,0]]).solve()
```

The model above should be unsat since [1,2,0] is a valid circuit, but it is sat for ortools and gurobi
It crashes with minizinc and z3 (because of the == 0 translation of the negation)

**Comment by Wout4** [comment]
The reason this doesn't work correctly is because the decomposition of Circuit generates auxiliary variables. When negating the decomposition the solver just changes the assignment of one of these variables, which has no effect on the original variables.

So this will be a problem for all global constraints that involve auxiliary variables (at the moment only Circuit)

**Comment by Dimosts** [comment]
This will (probably) be fixed by pull request #225

**Comment by Wout4** [comment]
Closed as fixed by #225
